### DISCUSSION TITLE
Compatibility with typed flavours of JS other than TypeScript? Like ReScript.

### DISCUSSION THREAD

**Discussion by redbar0n** [colinhacks/zod#928] [Q&A]
Just wondering how compatible Zod would be with something like ReScript?

Zod markets itself as pretty tied to TypeScript. From the Readme:

> TypeScript-first schema validation with static type inference 

> Zod is a TypeScript-first schema declaration and validation library.

> Zod will automatically infer the static TypeScript type.

But it also says:

> Works with plain JavaScript too! You don't need to use TypeScript.

So I'm a little uncertain if the type inference would work just as well with ReScript instead of TypeScript.

**Comment by scotttrinh** [comment]
> So I'm a little uncertain if the type inference would work just as well with ReScript instead of TypeScript.

I think the point here is that all TypeScript projects compile to JavaScript so you can use zod for runtime checking in JavaScript, too, but the type information that it emits is TypeScript. I _think_ there are tools to convert or use TypeScript `.d.ts` definition files for ReScript, but I'm not heavily involved in that ecosystem. Maybe something [like this](https://www.benadamstyles.com/blog/rescript-type-safe-bindings/) could work for you?

**Reply by redbar0n** [reply]
Thanks. As far as I understand, Zod doesn't actually emit type information (at either compile or runtime), other than make it easy for the [TS language server](https://code.visualstudio.com/api/language-extensions/overview) to perform type inference across functions and highlight errors? (In that case, [ReScript's language server](https://github.com/rescript-lang/rescript-vscode) should work...)

Otherwise, what would be the need for libraries such as these?

[ts-to-zod](https://github.com/fabien0102/ts-to-zod): Convert TypeScript definitions into Zod schemas.
[zod-to-ts](https://github.com/sachinraja/zod-to-ts): Generate TypeScript definitions from Zod schemas.

It's all very confusing how Zod does its magic...
